model/transformers/dffml_model_transformers/ner/ner_model.py

```diff
@@ -0,0 +1,994 @@
+        with tf.io.gfile.GFile(output_eval_file, "w") as writer:
+            for res in results:
+                for key, val in res.items():
+                    if "loss" in key:
+                        self.logger.info(key + " = " + str(val))
+                        writer.write(key + " = " + str(val))
+                        writer.write("\n")
+                    else:
+                        self.logger.info(key)
+                        self.logger.info("\n" + report)
+                        writer.write(key + "\n")
+                        writer.write(report)
+                        writer.write("\n")
+        # Return accuracy for the last checkpoint
+        return Accuracy(f1_score(y_true, y_pred))
+
+    async def predict(
+        self, records: AsyncIterator[Record]
+    ) -> AsyncIterator[Tuple[Record, Any, float]]:
+        if not os.path.isfile(
+            os.path.join(self.parent.config.output_dir, "tf_model.h5")
+        ):
+            raise ModelNotTrained("Train model before prediction.")
+        config = self.parent.config._asdict()
+        config["strategy"] = self.parent.config.strategy
+        config["n_device"] = self.parent.config.n_device
+        tokenizer = self.tokenizer_class.from_pretrained(
+            config["output_dir"], do_lower_case=config["do_lower_case"]
+        )
+        model = self.model_class.from_pretrained(config["output_dir"])
+        test_batch_size = (
+            config["per_device_eval_batch_size"] * config["n_device"]
+        )
+        async for record in records:
+            sentence = record.features([self.parent.config.WORDS.NAME])
+            df = pd.DataFrame(sentence, index=[0])
+            test_dataset, num_test_examples = self.get_dataset(
+                df,
+                tokenizer,
+                self.pad_token_label_id,
+                test_batch_size,
+                mode="test",
+            )
+            test_dataset = self.parent.config.strategy.experimental_distribute_dataset(
+                test_dataset
+            )
+            y_pred = self._custom_predict(
+                test_dataset,
+                tokenizer,
+                model,
+                num_test_examples,
+                test_batch_size,
+            )
+            preds = [
+                [
+                    {word: y_pred[i][j]}
+                    for j, word in enumerate(s.split()[: len(y_pred[i])])
+                ]
+                for i, s in enumerate(
+                    df[self.parent.config.WORDS.NAME].to_list()
+                )
+            ]
+            record.predicted(self.parent.config.predict.NAME, preds, "Nan")
+            yield record
+
+
+@entrypoint("ner_tagger")
+class NERModel(Model):
+    """
+    Implemented using HuggingFace Transformers Tensorflow based Models.
+
+    First we create the training and testing datasets
+
+    .. literalinclude:: /../model/transformers/examples/ner/train_data.sh
```

Comment:
It looks like we're missing these examples?

Reply:
Ohh Yes!!